### DISCUSSION TITLE
Hydrating vs updating?

### DISCUSSION THREAD

**Discussion by torian257x** [livewire/livewire#3584] [1. Help] [answered]
can someone explain what is the difference of `hydratingFoo` vs `updatingFoo` ? I assume, `hydratingFoo` involves water

https://laravel-livewire.com/docs/2.x/lifecycle-hooks

**Comment by jonneroelofs** [comment, marked as answer]
I think hydrating happens anytime you make a call from the browser. 
Updating only happens when you actually change the value of a property.

Let's say you have component: 
```php
class MyComponent extends \Livewire\Component {
    public $foo = 'bar';

    public function updatingFoo()
    {
        // I only get called when you change the value of $this->foo
    }
    
    public function hydratingFoo(){
        // I get called anytime you interact with this component
    }
    
    public function submit(){
        // when you call me I always hydrate
    }
    
}
```

If you now call submit with wire:click from your view, then that would trigger hydratingFoo but not updatingFoo
If you change an input with wire:model="foo" then that would call the updatingFoo hook.

And if you had 2 properties, then changing one would call the updating hook for that property but the hydration hook for both.

**Comment by joshhanley** [comment]
Yep what @jonneroelofs said is correct.

Every time a Livewire request is sent to the server, due to php being stateless, Livewire has to build up a component back to the state it was previously. It does that by keeping the state in the front end and sending it along with the request.

The process of building up the component class is called hydration. There is also dehydration, which is the opposite, where Livewire takes all current state from the component and serialises it to JSON and sends it to the front end.

So hydration methods are called whenever it is restoring state from a request, and dehydration is when its serialising state.

Update methods are only called in response to a field being modified by the front end, and they happen after hydration but before render and before dehydration.

Sometimes people think if they modify a property on the backend that the update methods will be called, but they don't. They only run if the changes comes from the front end.

There are some diagrams on this page, if you're interested https://laravel-livewire.com/docs/2.x/security

Hope this helps!
